wkcuber/recubing.py

```diff
@@ -0,0 +1,138 @@
+import logging
+import wkw
+import numpy as np
+from argparse import ArgumentParser
+
+from .utils import (
+    add_verbose_flag,
+    open_wkw,
+    WkwDatasetInfo,
+    ensure_wkw,
+    add_distribution_flags,
+    setup_logging,
+    infer_bounding_box,
+    BLOCK_LEN,
+)
+
+
+def create_parser():
+    parser = ArgumentParser()
+
+    parser.add_argument(
+        "source_path", help="Directory containing the datasource properties."
+    )
+
+    parser.add_argument(
+        "target_path", help="Output directory for the generated dataset."
+    )
+
+    parser.add_argument(
+        "--layer_name",
+        "-l",
+        help="Name of the cubed layer (color or segmentation)",
+        default="color",
+    )
+
+    parser.add_argument(
+        "--dtype",
+        "-d",
+        help="Target datatype (e.g. uint8, uint16, uint32)",
+        default="uint8",
+    )
+
+    parser.add_argument(
+        "--wkw_file_len", help="Target file length (default 32)", type=int, default=32
+    )
+
+    parser.add_argument(
+        "--no_compression",
+        help="Use compression, default false",
+        type=bool,
+        default=False,
+    )
+
+    add_verbose_flag(parser)
+    add_distribution_flags(parser)
+
+    return parser
+
+
+def next_lower_divisible_by(number, divisor) -> int:
+    remainder = number % divisor
+    return number - remainder
+
+
+def next_higher_divisible_by(number, divisor) -> int:
+    remainder = number % divisor
+    return number - remainder + divisor
+
+
+def recube(
+    source_path, target_path, layer_name, dtype, wkw_file_len=32, compression=True
+):
+    if compression:
+        block_type = wkw.Header.BLOCK_TYPE_LZ4
+    else:
+        block_type = wkw.Header.BLOCK_TYPE_RAW
+
+    target_wkw_header = wkw.Header(
+        np.dtype(dtype), file_len=wkw_file_len, block_type=block_type
+    )
+    target_wkw_info = WkwDatasetInfo(target_path, layer_name, 1, target_wkw_header)
+    source_wkw_header = wkw.Header(np.dtype(dtype))
+    source_wkw_info = WkwDatasetInfo(source_path, layer_name, 1, source_wkw_header)
+
+    ensure_wkw(target_wkw_info)
+
+    bounding_box = infer_bounding_box(source_wkw_info)
+    bottom_right = [
+        coord + size for coord, size in zip(bounding_box[0], bounding_box[1])
+    ]
+
+    wkw_cube_size = wkw_file_len * BLOCK_LEN
```

Comment:
Instead, of `BLOCK_LEN` is better to use `target_wkw_header.block_len` since it's the proper value (and not an imported, static number).